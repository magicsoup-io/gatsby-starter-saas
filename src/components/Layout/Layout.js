import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { StaticQuery, graphql } from 'gatsby'
import { ThemeProvider } from 'styled-components'

import Header from '../Header'
import Footer from '../Footer'
import { Box } from '@magicsoup.io/stock'

import theme from '../../styled/theme'
import GlobalStyle from '../../styled/global-styles'

const Layout = ({ children }) => (
  <ThemeProvider theme={theme}>
    <Fragment>
      <StaticQuery
        query={graphql`
          query SiteTitleQuery {
            site {
              siteMetadata {
                title
              }
            }
          }
        `}
        render={data => (
          <Fragment>
            <Header siteTitle={data.site.siteMetadata.title} />
            <Box 
              as='main' 
              py={6}>
              {children}
            </Box>
            <Footer />
          </Fragment>
        )}
      />
      <GlobalStyle />
    </Fragment>
  </ThemeProvider>
)

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Layout
