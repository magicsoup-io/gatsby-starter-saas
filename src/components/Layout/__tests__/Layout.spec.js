import React from "react"
import renderer from "react-test-renderer"
import { StaticQuery } from 'gatsby'
import Layout from "../Layout"

jest.mock('../../Header', () => ()=> <header id="mockHeader">
   mockHeader
</header>)

jest.mock('../../Footer', () => ()=> <footer id="mockFooter">
   mockFooter
</footer>)

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: 'Website Title',
        },
      },
    })
  )

})

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Layout>Some child</Layout>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})