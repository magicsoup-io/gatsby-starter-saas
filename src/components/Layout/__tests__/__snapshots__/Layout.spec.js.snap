// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <header
    id="mockHeader"
  >
    mockHeader
  </header>,
  <main
    className="sc-bdVaJa dzLjON"
  >
    Some child
  </main>,
  <footer
    id="mockFooter"
  >
    mockFooter
  </footer>,
]
`;
